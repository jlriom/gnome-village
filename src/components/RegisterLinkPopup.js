import React, { useState } from "react";
import { Nav } from "react-bootstrap";
import RegisterPage from "../containers/user/RegisterPage";

const RegisterLinkPopup = () => {
	const [show, setShow] = useState(false);
	return (
		<>
			<Nav.Link onClick={() => setShow(true)}>Register</Nav.Link>
			<RegisterPage
				show={show}
				handleClose={() => setShow(false)}></RegisterPage>
		</>
	);
};

export default RegisterLinkPopup;
