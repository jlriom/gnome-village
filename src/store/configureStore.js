import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
