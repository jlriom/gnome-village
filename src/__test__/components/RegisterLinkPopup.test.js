import React from "react";
import RegisterLinkPopup from "../../components/RegisterLinkPopup";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

import configureStore from "../../store/configureStore";
const store = configureStore();


describe("<RegisterLinkPopup />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <RegisterLinkPopup />
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});
