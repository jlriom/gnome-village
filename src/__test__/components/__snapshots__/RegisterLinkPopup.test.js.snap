// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterLinkPopup /> renders ok 1`] = `
<a
  className="nav-link"
  href="#"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
>
  Register
</a>
`;
